### DISCUSSION TITLE
How would you modify a TOC at build time?

### DISCUSSION THREAD

**Discussion by DavidMarquezF** [dotnet/docfx#9943] [Q&A]
In our docs we have some articles that should get excluded depending on a metadata field in the TOC items and a parameter passed to the cmd. From my understanding this should probably be achievable by using document processors. I was looking for a way to extend the TocDocumentProcessor or add another Toc build step, but I'm not sure which is the way to go.

Has anybody tried to do something similar or has any recommendations on how to approach it?

Thanks!

**Comment by filzrev** [comment]
It can customize generated TOCs on build-time by adding `toc.extension.js` to [docfx custom template](https://dotnet.github.io/docfx/tutorial/howto_create_custom_template.html).

> In our docs we have some articles that should get excluded depending on a metadata field in the TOC items and a parameter passed to the cmd.

Thought by this approach. it can access TOC items metadata. But can't read cmd parameters.

> [!NOTE]
> Currently `toc.extension.js` is applied to `toc.html` only and not applied to `toc.json` (that used by `modern` template).
> So it need to customize [toc.json.js](https://github.com/dotnet/docfx/blob/main/templates/default/toc.json.js) file also.
> See: https://github.com/dotnet/docfx/issues/9944#issuecomment-2132089192
